tests/test_flows.py

```diff
@@ -568,30 +578,23 @@ def test_channel_selection(self):
             )
             time.sleep(1)
 
-            debug(
-                "Confirming channel selection request has been received on agent,step {}".format(i))
-
-            self.check_log(env.agents[0].radios[0], "CHANNEL_SELECTION_REQUEST_MESSAGE")
-            self.check_log(env.agents[0].radios[1], "CHANNEL_SELECTION_REQUEST_MESSAGE")
-
-            debug(
-                "Confirming tlvTransmitPowerLimit has been received with correct value on \
-                    agent, step {}".format(i))
+            debug("Confirming tlvTransmitPowerLimit has been received with correct value on agent,"
+                  " step {}".format(i))
 
             self.check_log(env.agents[0].radios[0], "tlvTransmitPowerLimit {}".format(tp20dBm))
-
-            self.check_log(env.agents[0].radios[0],
-                           "ACTION_APMANAGER_HOSTAP_CHANNEL_SWITCH_ACS_START")
-
             self.check_log(env.agents[0].radios[1], "tlvTransmitPowerLimit {}".format(tp20dBm))
 
-            self.check_log(env.agents[0].radios[0],
```

Comment:
I think this was a bug in the test because the wrong log file was being checked. Nevertheless, the test was passing.
For the sake of curiosity: i think we were going to implement some kind of a cursor in the log. Is it available yet?

Reply:
> I think this was a bug in the test because the wrong log file was being checked. Nevertheless, the test was passing.

Yes, well, that's forever the problem when writing the test after the code: how do you know that the test is really testing what it should?

> For the sake of curiosity: i think we were going to implement some kind of a cursor in the log. Is it available yet?

It is available, but the interface is clunky (you have to store the line number in the check_log return value). The idea is to instead store the line number in the checkpoint() function. Note that the checkpoint() function is currently already invoked in this particular test. And note that that wouldn't have caught this issue, because we don't checkpoint between the two log checks (would be a bad idea to do that, since it forces you to rely on the order of logs).

Reply:
“Program testing can be used to show the presence of bugs, but never to show their absence!”
― Edsger W. Dijkstra